libs/application/template-api-modules/src/lib/modules/templates/reference-template/reference-template.module.ts

```diff
@@ -0,0 +1,22 @@
+import { DynamicModule } from '@nestjs/common'
+
+// This is a shared module that gives you access to common methods
+import { SharedTemplateAPIModule } from '../../shared'
+
+// The base config that template api modules are registered with by default
+// (configurable inside `template-api.module.ts`)
+import { BaseTemplateAPIModuleConfig } from '../../../types'
+
+// Here you import your module service
+import { ReferenceTemplateService } from './reference-template.service'
+
+export class ReferenceTemplateModule {
+  static register(config: BaseTemplateAPIModuleConfig): DynamicModule {
+    return {
+      module: ReferenceTemplateModule,
+      imports: [SharedTemplateAPIModule.register(config)],
```

Comment:
Maybe ConfigModule should be [global](https://docs.nestjs.com/techniques/configuration#use-module-globally), so all of the template modules can be static instead of dynamic.

Reply:
Will revisit in another PR